README.md

```diff
@@ -9,18 +9,66 @@
 [![pre-commit](https://github.com/jupyter-xeus/cpp-terminal/actions/workflows/pre-commit-check.yml/badge.svg)](https://github.com/jupyter-xeus/cpp-terminal/actions/workflows/pre-commit-check.yml)
 [![pre-commit.ci status](https://results.pre-commit.ci/badge/github/jupyter-xeus/cpp-terminal/master.svg)](https://results.pre-commit.ci/latest/github/jupyter-xeus/cpp-terminal/maset)
 
-`CPP-Terminal` is a small and simple library for writing platform independent terminal applications. It works on Windows, MacOS and Linux and offers a simple programming interface. It
-supports colors, keyboard input and has all the basic features to write any
-terminal application.
+`CPP-Terminal` is a small, simple and dependency-free library for writing platform independent terminal applications. It tries to follow the C++ 'Zero-overhead principle' and its header files only include std ones. It works on Windows, MacOS and
+Linux and offers a simple programming interface. It supports colors, keyboard input, terminal resizing and has all the basic features to write any terminal application. On windows the library automatically allocate a console on GUI application.
+
+## Minimal examples
+
+The minimal Hello World is as simple as :
+
+```cpp
+#include "cpp-terminal/io.hpp"
+
+#include <iostream>
+
+int main()
+{
+  std::cout << "\033[31mHello world !\033[0m" << std::endl;
```

Comment:
what happens on an exception? Does the terminal revert itself back properly?

Reply:
Please. I skipped grammar classes indeed 😂. French and english is really not a good mix

Reply:
@MCWertGaming If I remember well the C++ rules : 
```
If an exception is thrown and not caught, including exceptions that escape the initial function of [std::thread](https://en.cppreference.com/w/cpp/thread/thread), the main function, and the constructor or destructor of any static or thread-local objects, then [std::terminate](https://en.cppreference.com/w/cpp/error/terminate) is called. It is implementation-defined whether any stack unwinding takes place for uncaught exceptions.
``
So it's depends :(

Reply:
> Please. I skipped grammar classes indeed joy. French and english is really not a good mix

I can tell, was the same to me with french class :p


> @MCWertGaming If I remember well the C++ rules :
> 
>     * If an exception is thrown and not caught, including exceptions that escape the initial function of [std::thread](https://en.cppreference.com/w/cpp/thread/thread), the main function, and the constructor or destructor of any static or thread-local objects, then [std::terminate](https://en.cppreference.com/w/cpp/error/terminate) is called. It is implementation-defined whether any stack unwinding takes place for uncaught exceptions.
> 
> 
> So it's depends :(

Alrighty ^^